### PR TITLE
NH2: Display build info when initializing

### LINKED ISSUE #979: Add build date and/or git commit information to nethook console logs
People create issues that nethook does not work but do not specify which version they are using.

### PR DESCRIPTION
**Pull request by yaakov-h**
Example (dirty working tree):

image

Example (clean working tree):

image

Format is:

> Built at \<build time\> from \<commit sha\>[/dirty] (\<commit date\>)

Resolves #979.

### CODE REVIEW

File: Resources/NetHook2/GenerateVersionInfo.ps1 [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        If (Get-Command git) {
+            return $true
+        }
+    }
+    Catch {
+        return $false
+    }
+}
+
+$buildDate = Get-Date -Format "yyyy-MM-dd HH:mm:ss K"
```
Comment (xPaw) [inline Resources/NetHook2/GenerateVersionInfo.ps1:14]: Add `-AsUTC`?

File: Resources/NetHook2/GenerateVersionInfo.ps1 [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+if (Test-GitAvailable) {
```
Comment (xPaw) [inline Resources/NetHook2/GenerateVersionInfo.ps1:16]: Don't you want to add default/fallback variables if it happens to be not available?

Comment (yaakov-h) [reply Resources/NetHook2/GenerateVersionInfo.ps1:16]: the strings should become empty string, I need to test the bool.

File: Resources/NetHook2/GenerateVersionInfo.ps1 (hunk 1 above)
Comment (yaakov-h) [reply Resources/NetHook2/GenerateVersionInfo.ps1:14]: >Get-Date : A parameter cannot be found that matches parameter name 'AsUtc'.

**Comment by yaakov-h** [comment]
Build info from the DLL produced by GitHub Actions. The build from the PR was from `+ddc051dfbdbb49fedf79b659f72fef59e2de9336:refs/remotes/pull/980/merge`.

image

File: Resources/NetHook2/GenerateVersionInfo.ps1 (hunk 1 above)
Comment (xPaw) [reply Resources/NetHook2/GenerateVersionInfo.ps1:14]: Needs newer powershell I guess?

https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/get-date?view=powershell-7.1#example-10--return-a-date-value-interpreted-as-utc

Comment (yaakov-h) [reply Resources/NetHook2/GenerateVersionInfo.ps1:14]: Probably. PowerShell Core adds a bunch of new stuff... but also removes some stuff. It's also not preinstalled with Windows so we can't rely on it existing.

Comment (psychonic) [reply Resources/NetHook2/GenerateVersionInfo.ps1:14]: `[DateTime]::UtcNow.ToString('yyyy-MM-dd HH:mm:ss K')`

Comment (yaakov-h) [reply Resources/NetHook2/GenerateVersionInfo.ps1:14]: tbh I’d rather keep it in the system timezone unless you can also figure out how to get the git commit time in utc. I guess we could parse and reformat it?

Comment (xPaw) [reply Resources/NetHook2/GenerateVersionInfo.ps1:14]: Getting git commit time is pointless when you already have the commit.

**Review by xPaw — APPROVED** [review]

**Pull request merged**